.github/workflows/sonarcloud.yml

```diff
@@ -1,54 +1,76 @@
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v4
-
-    - name: Cache dependencies
-      id: cache-dependencies
-      uses: actions/cache@v3
-      with:
-        path: java/lib
-        key: ${{ runner.os }}-java-lib-${{ hashFiles('java/buildconf/ivy/*.*') }}
-
-    - name: Resolve dependencies
-      if: steps.cache-dependencies.outputs.cache-hit != 'true'
-      run: |
-        ant -f java/manager-build.xml ivy
-
-    - name: Compile Java
-      run: ant -f java/manager-build.xml compile
-
-    - name: SonarCloud Scan
-      uses: sonarsource/sonarcloud-github-action@master
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        SONAR_TOKEN: ${{ secrets.SONARQUBE_AUTH_UYUNI }}
-        args: >
-          -Dsonar.junit.reportPaths=""
-          -Dsonar.coverage.jacoco.xmlReportPaths=""
-          -Dsonar.pullrequest.key=${{ github.event.number }}
-          -Dsonar.pullrequest.branch=${{ github.event.pull_request.head.ref }}
-          -Dsonar.pullrequest.base=${{ github.base_ref }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.workflow_run.head_repository.full_name }}
+          ref: ${{ github.event.workflow_run.head_branch }}
+
+      - name: Download Java built artifact
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "java-built-files"
+            })[0];
+            let download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            let fs = require('fs');
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/java-built-files.zip`, Buffer.from(download.data));
+
+      - name: Unzip java built artifact
+        run: |
+          unzip java-built-files.zip
+
+      - name: Cache dependencies
+        id: cache-dependencies
+        uses: actions/cache@v3
+        with:
+          path: java/lib
+          key: ${{ runner.os }}-java-lib-${{ hashFiles('java/buildconf/ivy/*.*') }}
+
+      - name: Resolve dependencies
+        if: steps.cache-dependencies.outputs.cache-hit != 'true'
+        run: |
+          docker run -v $PWD:/src -w /src \
+            registry.opensuse.org/systemsmanagement/uyuni/master/docker/containers/uyuni-master-pgsql:latest \
+            ant -f java/manager-build.xml ivy
```

Comment:
TODO:
We need to move this to the checkstyle action.
Probably sharing this content on the same artifact that we upload.